src/parsers/manifestjson.js

```diff
@@ -146,6 +150,17 @@ export default class ManifestJSONParser extends JSONParser {
     }
   }
 
+  validateIcons() {
+    const { icons } = this.parsedJSON;
+    Object.keys(icons).forEach((size) => {
+      const path = icons[size];
+      if (!this.io.files.hasOwnProperty(path)) {
+        this.collector.addError(messages.manifestIconMissing(path));
+        this.isValid = false;
+      }
+    });
+  }
```

Comment:
I filed some follow up issues to do that. #1328 to check that they're valid images and #1329 to ensure they're reasonable sizes.

As long as Firefox can render the image the file type might not matter, but that seems reasonable.